pkg/kubelet/kubelet.go

```diff
@@ -550,24 +554,26 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	var secretManager secret.Manager
 	var configMapManager configmap.Manager
-	switch kubeCfg.ConfigMapAndSecretChangeDetectionStrategy {
-	case kubeletconfiginternal.WatchChangeDetectionStrategy:
-		secretManager = secret.NewWatchingSecretManager(kubeDeps.KubeClient, klet.resyncInterval)
-		configMapManager = configmap.NewWatchingConfigMapManager(kubeDeps.KubeClient, klet.resyncInterval)
-	case kubeletconfiginternal.TTLCacheChangeDetectionStrategy:
-		secretManager = secret.NewCachingSecretManager(
-			kubeDeps.KubeClient, manager.GetObjectTTLFromNodeFunc(klet.GetNode))
-		configMapManager = configmap.NewCachingConfigMapManager(
-			kubeDeps.KubeClient, manager.GetObjectTTLFromNodeFunc(klet.GetNode))
-	case kubeletconfiginternal.GetChangeDetectionStrategy:
-		secretManager = secret.NewSimpleSecretManager(kubeDeps.KubeClient)
-		configMapManager = configmap.NewSimpleConfigMapManager(kubeDeps.KubeClient)
-	default:
-		return nil, fmt.Errorf("unknown configmap and secret manager mode: %v", kubeCfg.ConfigMapAndSecretChangeDetectionStrategy)
-	}
-
-	klet.secretManager = secretManager
-	klet.configMapManager = configMapManager
+	if klet.kubeClient != nil {
```

Comment:
note: klet.kubeClient is equal to kubeDeps.KubeClient (set above on line 502)

Reply:
yes.

https://github.com/kubernetes/kubernetes/pull/113501#discussion_r1011358876

- After `klet` is set, I changed to use `klet.kubeClient` here. 
- Before `klet` is set, we use `kubeDeps.KubeClient`

Reply:
This is per @bart0sh 's comment.
- I think this is clear and will help if we will have some refactoring on this later.